### ISSUE TITLE
[18GB] Allow options to be selected even if the player count is a range instead of an exact match.

### ISSUE THREAD

**Issue by slapphappyjazz** [tobymao/18xx#8579]
When creating a game of 18GB, there are options for alternative set ups for the 2-player game and the 4-player game.  Currently, you can only select these if you are creating a game for exactly 2 or 4 players.  I think you should be able to select these options if the number of players is a range that includes 2 or 4, rather than requiring an exact match.

E.g. I created a game and wanted to allow either 3 or 4 players to play.  I also wanted the alternative 4-player setup if the game had 4 players.  The UI wouldn't let me select this option.

**Comment by andypymont** [comment]
I think this is potentially an issue/UI direction discussion for all games where there are variants which are limited by player count, not just 18GB.

Before #8133, a game was created with a maximum player count but not with a minimum. At the time, picking a player-count-specific variant would effectively enforce a minimum/lock the game to a specific player count. (If no player-count-specific variant was selected, the game would instead have the title's minimum player count, the specified maximum, and otherwise remain flexible.)

Now ranges have been implemented, so the game creator can specify both minimum and maximum. In order to activate the player-count-specific variant, they need to set both minimum and maximum the same. This effectively enforces the same behaviour as pre-#8133, except of course it is now much more obvious in the user interface that this is what is happening.

For 18GB I think what Dave has requested here would work perfectly fine, i.e. have the alternative 4P setup specified, but only use it if the game is actually launched with exactly 4 players. But I don't know all the other titles on the site well enough to comment on whether changing variants to not limit the pre-game player count, but still apply when the launch player count matches, is going to work. Will this confuse some players because they select a variant which then ends up being "dropped" if their eventual game-start player count is not compatible with the variant?

(I'm relatively busy in December; when I have the chance I can look at which other player-count-specific variants already exist across all games so we can evaluate the impact of potentially changing this. In the meantime I'd welcome comments from others on whether this would be confusing for 18GB and for any other games with variants which apply at specific player counts, based on their own experience.)

**Comment by andypymont** [comment]
This is what this change would look like - does this make sense? Or would it be confusing to see e.g. both variants listed when one or neither of them will actually end up being used for the game?

![image](https://user-images.githubusercontent.com/8403378/210250982-b3dd6ecd-8b6d-44a9-a4b7-d74491867ca7.png)
![image](https://user-images.githubusercontent.com/8403378/210250923-f424da4d-628d-43f6-b0c9-fcefdc7ab8c2.png)

**Comment by slapphappyjazz** [comment]
I think that would make sense.   It gives people the right information about what they would be signing up for.  I'd say go with it.

Would this affect other titles too?

**Comment by andypymont** [comment]
> I think that would make sense. It gives people the right information about what they would be signing up for. I'd say go with it.
> 
> Would this affect other titles too?

The draft I have ready at the moment is just for 18GB. I see that 18FL and 18VA actually already have the behaviour that Dave is suggesting here - there is a 2p-specific variant on each, which can be turned on whatever the player count but only impacts 2-player games. So we could move 18GB to the same standard as these two games.

I don't know the other games with player-count-specific variants well enough to know how this would work for them and whether it would be confusing for players setting up games, so I don't feel qualified to make a general design decision about how variants should be handled.

The other games with variants which are currently restricted to specific player counts are:

- 18LosAngeles - "Dewey, Cheatham, and Howe" and "LA Title"
- 18Scan - "70% Corporation Holding Limit"
- 18Tokaido - "Pass Priority" and "No Discard"
- 1825 - numerous variants/options
- 1840 - "3P Small Map"
- 1888 - "North Map"

**Comment by slapphappyjazz** [comment]
Of the others you list, I'd definitely like to see this behaviour for 1840 - i.e. select a player range and say that if the game starts with three players, it will use the three-player map.
1825 is different and requires dedicated code.  (I'd actually implement it the other way around; let the creator select which units they want to use and restrict the player count accordingly).   
1888 only has one map at present.  I guess it will end up being like 1825, with different combinations of maps to choose from.
18 Los Angeles seems to allow the options for any number of players except the two-player game.  I think the 18GB/18FL/18VA behaviour would work for 18 Los Angeles as well.
I don't know 18 Tokaido; it could do with labelling its options with the player counts they apply to, for clarity (I think one is 2-3 players and another is 3-4 players).  My guess is that the same approach would work here too.
I can't see 18Scan on the list?

In summary, this behaviour seems a reasonable default to me but 1825 and possibly 1888 would be exceptions.  Of course, someone would need to check with the relevant designers and implementers. 
In the meantime, I'd be pleased if we move 18GB to the same standard as 18FL and 18VA.
